src/libraries/System.Text.Json/src/System/Text/Json/Reader/JsonReaderHelper.Unescaping.cs

```diff
@@ -236,6 +237,71 @@ public static string TranscodeHelper(ReadOnlySpan<byte> utf8Unescaped)
             }
         }
 
+        public static int TranscodeHelperWithCleanupOnFailure(ReadOnlySpan<byte> utf8Unescaped, Span<char> destination)
+        {
+            try
+            {
+#if BUILDING_INBOX_LIBRARY
+                return s_utf8Encoding.GetChars(utf8Unescaped, destination);
+#else
+                unsafe
+                {
+                    fixed (byte* srcPtr = utf8Unescaped)
+                    fixed (char* destPtr = destination)
+                    {
+                        return s_utf8Encoding.GetChars(srcPtr, utf8Unescaped.Length, destPtr, destination.Length);
+                    }
+                }
+#endif
+            }
+            catch (DecoderFallbackException dfe)
+            {
+                destination.Clear();
```

Comment:
Why do we want to clear it?

Reply:
Good question. Consensus during API review was that the destination buffer should be cleared if it's too short, but it might not be too important given that we're not surfacing a `TryCopyString` variant (we might add one in the future given demand).

This case however specifically concerns clearing the buffer during decode errors, which is a different error mode from the buffer being too short. I'm clearing the buffer in that case as well since it was easy to add, but don't have a strong opinion otherwise.